run_analysis/run_analysis.R

```diff
@@ -158,53 +216,81 @@ normalise_c_id_power_by_daily_max <- function(combined_data, max_power, pre_even
-                                    is.na(manufacturer))
-    manufacturer_install_data <- filter(manufacturer_install_data, manufacturer != "SolarEdge" | 
-                                          is.na(manufacturer))
+upscale_and_summarise_disconnections <- function(circuit_summary,
+                                                 manufacturer_install_data,
+                                                 load_date,
+                                                 region_to_load,
+                                                 exclude_solar_edge) {
+  logdebug('Summarise and upscale disconnections on a manufacturer basis.', logger = logger)
+  if (exclude_solar_edge) {
+    circuits_to_summarise <- filter(circuit_summary, manufacturer != "SolarEdge" | is.na(manufacturer))
+    manufacturer_install_data <- filter(manufacturer_install_data, manufacturer != "SolarEdge" | is.na(manufacturer))
   } else {
     circuits_to_summarise <- circuit_summary
     manufacturer_install_data <- manufacturer_install_data
   }
-  upscaling_results <- get_upscaling_results(circuits_to_summarise, manufacturer_install_data, load_date, 
-                                              region_to_load, sample_threshold = 30)
+  upscaling_results <- get_upscaling_results(
+    circuits_to_summarise,
+    manufacturer_install_data,
+    load_date,
+    region_to_load,
+    sample_threshold = 30
+  )
   upscaling_results$with_separate_ufls_counts <- get_upscaling_results_excluding_ufls_affected_circuits(
-    circuits_to_summarise, manufacturer_install_data, load_date, region_to_load, sample_threshold = 30)
-
-  write.csv(upscaling_results$manufacturers_missing_from_cer, 
-            "logging/manufacturers_missing_from_cer.csv", row.names=FALSE)
-  write.csv(upscaling_results$manufacturers_missing_from_input_db, 
-            "logging/manufacturers_missing_from_input_db.csv", row.names=FALSE)
+    circuits_to_summarise,
+    manufacturer_install_data,
+    load_date,
+    region_to_load,
+    sample_threshold = 30
+  )
 
+  write.csv(
+    upscaling_results$manufacturers_missing_from_cer,
+    "logging/manufacturers_missing_from_cer.csv",
+    row.names = FALSE
+  )
+  write.csv(
+    upscaling_results$manufacturers_missing_from_input_db,
+    "logging/manufacturers_missing_from_input_db.csv",
+    row.names = FALSE
+  )
   return(upscaling_results)
 }
 
 #' Check if grouping is occuring
 #' @return boolean no_grouping, TRUE when no grouping is occuring
 check_grouping <- function(settings) {
-  if (settings$standard_agg==FALSE & settings$pst_agg==FALSE & settings$grouping_agg==FALSE &
-      settings$manufacturer_agg==FALSE & settings$model_agg==FALSE & settings$zone_agg==FALSE &
-      settings$circuit_agg==TRUE & settings$compliance_agg==TRUE & settings$compliance_2020_agg==TRUE
-      & settings$reconnection_compliance_agg & settings$v_excursion_agg==FALSE){
-    no_grouping=TRUE
+  if (
```

Comment:
I did notice that the style guide says use '&&' instead of '&' in if statements. Is this something worth changing?

Reply:
Yes, in this case it definitely makes sense to change '&' to '&&'. However, I just tried changing a bunch of '&' to '&&' and over-corrected, so I ended up with a bunch of errors